src/compiler/checker.ts

```diff
@@ -5750,6 +5755,25 @@ namespace ts {
                 }
                 return Ternary.False;
             }
+
+            function enumRelatedTo(source: Type, target: Type) {
+                if (source.symbol.name !== target.symbol.name) {
+                    return Ternary.False;
+                }
+                const sourceDecl = <EnumDeclaration>getMergedSymbol(source.symbol).valueDeclaration;

```

Comment:
using `valueDeclaration` does not look correct given that enums are open-ended and can be merged